src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/vilkårsvurdering/validering/BarnetsAlderVilkårValidator.kt

```diff
@@ -16,8 +17,17 @@ class BarnetsAlderVilkårValidator(
     fun validerVilkårBarnetsAlder(
         perioder: List<IkkeNullbarPeriode<VilkårResultat>>,
         barn: Person,
+        skalBrukeNyValideringForAdopsjonsbarn: Boolean = true,
     ): List<String> {
-        val vilkårLovverkInformasjonForBarn = VilkårLovverkInformasjonForBarn(barn.fødselsdato)
+        val vilkårLovverkInformasjonForBarn =
+            if (perioder.any { it.verdi.erAdopsjonOppfylt() } && skalBrukeNyValideringForAdopsjonsbarn) {
+                val fomAdoptertBarn = perioder.minOf { it.fom }.toYearMonth()
+                val tomAdoptertBarn = perioder.maxOf { it.tom }.toYearMonth()
+
+                VilkårLovverkInformasjonForBarn(fødselsdato = barn.fødselsdato, periodeFomForAdoptertBarn = fomAdoptertBarn, periodeTomForAdoptertBarn = tomAdoptertBarn)
```

Comment:
Dette er den faktiske endringen. Sørger for at hvis det gjelder adopsjon så sender vi med fom og tom, slik at man ikke bare tar utgangspunkt i fødselsdato.